src/commonMain/kotlin/org/mongodb/kbson/serialization/Serializers.kt

```diff
@@ -68,11 +74,168 @@ import org.mongodb.kbson.BsonTimestamp
 import org.mongodb.kbson.BsonType
 import org.mongodb.kbson.BsonUndefined
 import org.mongodb.kbson.BsonValue
+import org.mongodb.kbson.ExperimentalApi
 import org.mongodb.kbson.internal.Base64Utils
 import org.mongodb.kbson.internal.HexUtils
 import org.mongodb.kbson.internal.validateSerialization
 
-/** The Bson companion object */
+@ExperimentalApi
+internal fun <T> Ejson.writeBson(value: T, serializer: SerializationStrategy<T>): BsonValue {
+    if (value is BsonValue) return value
+    lateinit var result: BsonValue
+    val encoder = PrimitiveBsonEncoder(serializersModule) { result = it }
+    encoder.encodeSerializableValue(serializer, value)
+    return result
+}
+
+@ExperimentalApi
+internal fun <T> Ejson.readBson(element: BsonValue, deserializer: DeserializationStrategy<T>): T =
+    BsonDecoder(element, serializersModule, ignoreUnknownKeys).decodeSerializableValue(deserializer)
+
+
+/**
+ * Serializes the given [value] into an equivalent [JsonElement] using a serializer retrieved
+ * from reified type parameter.
+ *
+ * @throws [SerializationException] if the given value cannot be serialized to JSON.
```

Comment:
Did you mean JSON or BSON here? I would assume BSON since that is the output?

Reply:
BSON